### PR TITLE
Add preprocess module (1st step)

### PR DESCRIPTION
**Pull request by roshankern**
This PR adds the first step in the preprocess module. We add the env, lib, rule, and script files necessary for this first step. We also add a `pyproject.toml` file to make the python file accessible.

@mat10d this is ready for review when you get a chance!

*Note*: I opted to make the first step output TSVs instead of pickles, but this is easy to revert if we would rather stick with pickles.

### CODE REVIEW

File: workflow/scripts/preprocess/extract_metadata_tile.py
Comment (mat10d) [inline workflow/scripts/preprocess/extract_metadata_tile.py:1]: Super clean. Nice

File: workflow/rules/preprocess.smk
Comment (mat10d) [inline workflow/rules/preprocess.smk:1]: Cool--so this is the main preprocess script

**Comment by mat10d** [comment]
Ok, so just to summarize how this works--correct me if I'm wrong.

In example analysis, you run a bash script, which calls Snakefile, which calls the preprocess.smk, which calls the functions in the preprocess.py. Is that right?

**Review by mat10d — APPROVED** [review]

File: workflow/lib/preprocess/file_utils.py
[the remark this answers is not in the document]
Comment (roshankern) [reply workflow/lib/preprocess/file_utils.py:1]: Yep! This is actually novel code for handling the new system with file paths stored in the config tsv files.

File: workflow/rules/preprocess.smk
Comment (roshankern) [reply workflow/rules/preprocess.smk:1]: Yep!

**Comment by roshankern** [comment]
> In example analysis, you run a bash script, which calls Snakefile, which calls the preprocess.smk, which calls the functions in the preprocess.py. Is that right?

I think you understand but I want to clarify the last part for this step in particular:
`example_analysis/run_analysis.sh` -> `workflow/Snakefile` -> `workflow/rules/preprocess.smk` -> `workflow/scripts/preprocess/extract_metadata_tile.py` -> helper code in `workflow/lib/preprocess/`

**Pull request merged**